### PR TITLE
feat(STN-770): Gradient hero uniform height update to carousel JS

### PR DESCRIPTION
**Pull request by rkelbel**
# READY FOR REVIEW

## Summary
As a site visitor, I would like the Hero (Gradient) - Slider to have a uniform height as I scroll through the slides.

JIRA Card: https://sparkbox.atlassian.net/browse/STN-770

## Need Review By (Date)
5/26

## Urgency
medium

## Steps to Test
1. npm run test
2. npm run build:js
2. Create a Flexible page with multiple sliders, both "Hero Layered Sliders" and "Gradient Hero Sliders" for uniform behavior testing.
3. Enter different amounts of text content to each slide, one of them having a very large maxed out Title text, that gives the largest amount of space to the slide height for it to be very noticeable.
4. View your page and swap through the slides on each slider.

## Verification:
- [x] Each slider runs individual of the others with height, the slides within the individual slider should control the overall height.
- [x] Confirm there is no "jumpiness" when sliding through the slides.
- [x] There should be no new console errors
- [x] All content is accessible when no-js is enabled
- [x] If a page does not contain a Hero (Gradient) - Slider the JS should not execute
- [x] Confirm this works for both logged in and logged out users consistently

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)

### CODE REVIEW

File: docroot/themes/humsci/humsci_basic/src/js/shared/carousel-slides/carousel-slides-height.js [hunk 1]
```diff
@@ -1,18 +1,20 @@
-const slides = document.querySelectorAll('.paragraph--type--hs-carousel');
+// This work below applies uniform height to both the Hero Layered Slider (formerly Carousel) and 
+// the Hero Gradient Slider paragraph component slides.
+const slides = document.querySelectorAll('.paragraph--type--hs-carousel, .ptype-hs-gradient-hero-slider');
```
Comment (heatherdesigns) [inline docroot/themes/humsci/humsci_basic/src/js/shared/carousel-slides/carousel-slides-height.js]: The class of `ptype-hs-gradient-hero-slider` only targets the sliders within the main content area. Can we update this to `paragraph--type--hs-gradient-hero-slider` which wraps the sliders in the hero and main content areas?

**Review by heatherdesigns — COMMENTED** [review]
Fantastic work!! 

rexcellent

I'm so happy the JS for the carousel works seamlessly for the hero gradient slides as well! I left one recommendation in the code to update a class that will allow all gradient hero sliders to be queried by the JS.

---

For the gradient heros, I think we may want to consider centering the content within `hb-gradient-hero__text`. With the controls centered, the content would look better centered as well.

center please

---

We should take advantage of the `no-js` class and hide the slider controls when JS is disabled. This is probably beyond the scope of the work in this card, so let's share this feedback during stand up and see if they would like for a new card to be created in the backlog.

hide icons with no-js class

File: docroot/themes/humsci/humsci_basic/src/js/shared/carousel-slides/carousel-slides-height.js (hunk 1 above)
Comment (heatherdesigns) [reply docroot/themes/humsci/humsci_basic/src/js/shared/carousel-slides/carousel-slides-height.js]: Regarding the centering of the text mentioned above, this wasn't addressed in the implementation or acceptance criteria within the JIRA card that scoped the work in this PR, so we should let H&S aware and then address in future work should they desire the content to be centered.

Comment (rkelbel) [reply docroot/themes/humsci/humsci_basic/src/js/shared/carousel-slides/carousel-slides-height.js]: This all makes sense to me, the class has been updated as suggested.

I have an idea of what should be added if H&S were to want the content centered, so we can handle that once we've discussed with them.

**Review by heatherdesigns — APPROVED** [review]

**Pull request merged**